packages/docusaurus-plugin-content-docs/src/sidebars/processor.ts

```diff
@@ -56,48 +63,71 @@ function toSidebarItemsGeneratorVersion(
 // Handle the generation of autogenerated sidebar items and other post-processing checks
 async function processSidebar(
   unprocessedSidebar: NormalizedSidebar,
-  {
+  params: SidebarProcessorParams,
+): Promise<Sidebar> {
+  const {
     sidebarItemsGenerator,
     numberPrefixParser,
     docs,
     version,
-    options,
-  }: SidebarProcessorProps,
-): Promise<Sidebar> {
+    sidebarOptions,
+  } = params;
+
   // Just a minor lazy transformation optimization
   const getSidebarItemsGeneratorDocsAndVersion = memoize(() => ({
     docs: docs.map(toSidebarItemsGeneratorDoc),
     version: toSidebarItemsGeneratorVersion(version),
   }));
 
-  async function handleAutoGeneratedItems(
+  async function processCategoryItem(
+    item: NormalizedSidebarItemCategory,
+  ): Promise<SidebarItemCategory> {
+    return {
+      ...item,
+      items: (await Promise.all(item.items.map(processItem))).flat(),
+    };
+  }
+
+  async function processAutoGeneratedItem(
+    item: SidebarItemAutogenerated,
+  ): Promise<SidebarItem[]> {
+    // TODO the returned type can't be trusted in practice (generator can be user-provided)
+    const generatedItems = await sidebarItemsGenerator({
+      item,
+      numberPrefixParser,
+      defaultSidebarItemsGenerator: DefaultSidebarItemsGenerator,
+      ...getSidebarItemsGeneratorDocsAndVersion(),
+      options: sidebarOptions,
+    });
+    // TODO validate generated items: user can generate bad items
```

Comment:
Maybe we should autogenerate at the very first of sidebar loading? I know category shorthands / unnormalized items make it very hard to parse, but we either validate in two go or we re-order the steps.

Reply:
The processor basically only handles the the auto-generated items so moving this in sidebar logic would just merge the processor/generator with sidebar loading logic.

The question is rather, should we generate before or after other steps (validate, normalize). It does not matter much atm, we can do both and there's no good reason to change that. 

If we generate earlier, a custom generator function can still output bad items and we should still validate those anyway.

I added this TODO because I spotted some missing validation in the current implementation that we could add as an low-value improvement (considering most users use the default generator, or they really know what they are doing). I don't plan to implement or refactor this in the current PR.